server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1239,6 +1277,21 @@ public void invariant() {
             assert singleNodeClusterChecker == null || (mode == Mode.LEADER && applierState.nodes().size() == 1)
                 : "Single node checker must exist iff there is a single-node cluster";
 
+            // clusterFormationClusterStateView is unsynchronised to unblock the health reporting from depending on the mutex
+            // However, since we are inside the mutex now, we expect it have been synchronised
+            ClusterFormationFailureHelper.ClusterFormationClusterStateView synchronisedClusterFormationClusterStateView =
+                new ClusterFormationFailureHelper.ClusterFormationClusterStateView(
+                    lastAcceptedClusterState.nodes().getLocalNode(),
+                    lastAcceptedClusterState.nodes().getMasterNodes(),
+                    lastAcceptedClusterState.version(),
+                    lastAcceptedClusterState.term(),
+                    lastAcceptedClusterState.getLastAcceptedConfiguration(),
+                    lastAcceptedClusterState.getLastCommittedConfiguration(),
+                    getCurrentTerm()
+                );
```

Comment:
Can we just `new ClusterFormationFailureHelper.ClusterFormationClusterStateView(lastAcceptedClusterState, getCurrentTerm())` here?